common/src/hash.rs

```diff
@@ -136,14 +137,15 @@ pub fn hash_float(value: f64) -> PyHash {
     };
     x = ((x << e) & MODULUS) | x >> (BITS32 - e);
 
-    fix_sentinel(x as PyHash * value.signum() as PyHash)
+    Some(fix_sentinel(x as PyHash * value.signum() as PyHash))
 }
 
-pub fn hash_complex(value: &Complex64) -> PyHash {
-    let re_hash = hash_float(value.re);
-    let im_hash = hash_float(value.im);
+#[inline]
+pub fn hash_complex(value: &Complex64) -> Option<PyHash> {
+    let re_hash = hash_float(value.re)?;
+    let im_hash = hash_float(value.im)?;
```

Comment:
`re_hash` and `im_hash` need not be propagated. They need to have a value of hash_pointer.

Reply:
I'm sorry for my misunderstanding. Could you check new implementation is correct again?